test/test_package.cpp

```diff
@@ -1065,15 +1060,57 @@ TEST(DDPackageTest, CloseToIdentity) {
+    static constexpr std::size_t STOCHASTIC_CACHE_OPS           = 1U;
+};
+
+using DensityMatrixPackageTest = dd::Package<DensityMatrixSimulatorDDPackageConfigTesting::UT_VEC_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::UT_VEC_INITIAL_ALLOCATION_SIZE,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::UT_MAT_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::UT_MAT_INITIAL_ALLOCATION_SIZE,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_VEC_ADD_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_ADD_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_TRANS_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_CONJ_TRANS_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_VEC_MULT_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_MAT_MULT_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_VEC_KRON_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_MAT_KRON_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_VEC_INNER_PROD_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_DM_NOISE_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::UT_DM_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::UT_DM_INITIAL_ALLOCATION_SIZE,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_DM_DM_MULT_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::CT_DM_ADD_NBUCKET,
+                                             DensityMatrixSimulatorDDPackageConfigTesting::STOCHASTIC_CACHE_OPS>;
+
 TEST(DDPackageTest, dNodeMultiply) {
     //Multiply dNode with mNode (MxMxM)
     dd::Qubit nr_qubits = 3;
-    auto      dd        = std::make_unique<dd::DensityMatrixSimulatorDDPackage>(nr_qubits);
+    auto      dd        = std::make_unique<DensityMatrixPackageTest>(nr_qubits);
     // Make zero density matrix
-    auto state = dd::dEdge::one;
-    for (dd::Qubit p = 0; p < 3; p++) {
-        state = dd->makeDDNode(p, std::array{state, dd::dEdge::zero, dd::dEdge::zero, dd::dEdge::zero});
-    }
+    auto state = dd->makeZeroDensityOperator();
     dd->incRef(state);
     std::vector<dd::mEdge> operations = {};
     operations.push_back(dd->makeGateDD(dd::Hmat, nr_qubits, 0));
```

Comment:
use `emplace_back` especially in cases where you create a temporary object to avoid the copy typically involved in `push_back`
also check for all other calls in this file.

Reply:
done